### PR TITLE
initial commit for parties schema

### PR DESCRIPTION
**Pull request by peter-freshworks**
*Issue #:* /bcgov/entity#2671

*Description of changes:*
Initial commit for parties schema 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-schemas license (Apache 2.0).

### CODE REVIEW

File: src/registry_schemas/schemas/party.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                },
+                "middleInitial": {
+                    "type": "string",
+                    "maxLength": 30
+                },
+                "orgName": {
+                    "type": "string",
+                    "maxLength": 50
+                },
+                "orgType": {
```
Comment (cameron-freshworks) [inline src/registry_schemas/schemas/party.json]: I wasn't aware of this property. What is this used for? Is this something we should be putting into the filing?

File: src/registry_schemas/schemas/party.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                        ]
+                    }
+                }
+            },
+            "required": [
+                "person",
+                "roles",
+                "mailingAddress",
+                "deliveryAddress",
+                "appointmentDate"
```
Comment (cameron-freshworks) [inline src/registry_schemas/schemas/party.json]: This is currently passed from the front end. Is this a requirement as well? The schema says yes..

File: src/registry_schemas/schemas/party.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                "deliveryAddress": {
+                    "$ref": "https://bcrs.gov.bc.ca/.well_known/schemas/address"
+                },
+                "mailingAddress": {
+                    "$ref": "https://bcrs.gov.bc.ca/.well_known/schemas/address"
+                },
+                "title": {
+                    "type": "string"
+                },
+                "appointmentDate": {
```
Comment (lekshmimallika-aot) [inline src/registry_schemas/schemas/party.json]: What will be the value of this? current date?

File: src/registry_schemas/example_data/schema_data.py [hunk 4]
```diff
@@ -794,10 +793,72 @@
...
+                'streetAddressAdditional': '',
+                'addressCity': 'delivery_address city',
+                'addressCountry': 'CA',
+                'postalCode': 'H0H0H0',
+                'addressRegion': 'BC'
+            },
+            'appointmentDate': '2018-01-01',
+            'roles': [
+                'Incorporator',
+                'Director'
```
Comment (lekshmimallika-aot) [inline src/registry_schemas/example_data/schema_data.py]: There is a role called 'Completing Party' as well in the UI

File: src/registry_schemas/schemas/party.json [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                        "null"
+                    ],
+                    "format": "date"
+                },
+                "roles": {
+                    "type": "array",
+                    "items": {
+                        "enum": [
+                            "Director",
+                            "Incorporator"
```
Comment (lekshmimallika-aot) [inline src/registry_schemas/schemas/party.json]: There is a role called 'Completing Party'

Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: hmmm, I added roles based on the legal-api db refactor...I shall look into this!

File: src/registry_schemas/schemas/party.json (hunk 3 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: good question, I added this based on the refactor work...maybe I can just remove it as "required" for now

File: src/registry_schemas/schemas/party.json (hunk 5 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: I do not see a "Completing Party" role in COLIN...will wait for Kial

File: src/registry_schemas/schemas/party.json (hunk 3 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: This should be set from the front end, and is the same this as appointmentDate in the current director object

File: src/registry_schemas/schemas/party.json (hunk 2 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: yes it is required from the front end

File: src/registry_schemas/schemas/party.json (hunk 1 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: This is to differentiate between a person and an organisation

File: src/registry_schemas/schemas/party.json (hunk 3 above)
Comment (lekshmimallika-aot) [reply src/registry_schemas/schemas/party.json]: The UI does not capture any date values in step 2. We can still pass the value as the current date from UI. But I am not sure whether this is relevant for type Org. So removing 'required' sounds like a good idea to me

File: src/registry_schemas/schemas/party.json (hunk 1 above)
Comment (cameron-freshworks) [reply src/registry_schemas/schemas/party.json]: Isn't that what `partyType` is?

File: src/registry_schemas/schemas/party.json (hunk 3 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: should be resolved :)

File: src/registry_schemas/schemas/party.json (hunk 1 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: should be resolved

File: src/registry_schemas/schemas/party.json (hunk 5 above)
Comment (peter-freshworks) [reply src/registry_schemas/schemas/party.json]: Added it...don't see it causing any harm

**Pull request merged**